imperial.js

```diff
@@ -1,20 +1,62 @@
-import { Nation } from "./constants.js";
+import { Nation, Strings, NationCard } from "./constants.js";
 import Action from "./action.js";
 
+Map.fromIterable = (iter, getValue) => {
+  const m = new Map();
+  for (const k of iter) {
+    m.set(k, getValue(k));
+  }
+  return m;
+};
+
+Array.fromIterable = (iter, getValue) => {
```

Comment:
I _think_ I like these `fromIterable` helpers but I wonder if they should go in a different file? Maybe a `helpers` file since that seems to me like what they are? Or maybe it's premature to move them out now.